lib/tasks/ci.rake

```diff
@@ -0,0 +1,97 @@
+require 'json'
+
+namespace :ci do
+  # WARNING: this modifies ruby source code to assign tags to specs.
+  # After running this, you may need to update the matrix in rails_tests.yaml for new buckets
+  # rails ci:update_spec_tags[/tmp/rspec_profiles]
+  # profile information is generated from: rspec --format json --out rspec_results.json
+  # or profile information cab be downloaded from gh actions archives
+  desc 'Update RSpec test tags based on profiling data'
+  task :update_spec_tags, [:profile_dir] => :environment do |_t, args|
+    profile_dir = args[:profile_dir] || 'rspec_profiles'
+
+    unless Dir.exist?(profile_dir)
+      puts "Profile dir '#{profile_dir}' not found."
+      exit 1
+    end
+
+    profile_groups = []
+    Dir.foreach(profile_dir) do |filename|
+      next unless filename.end_with?('.json')
+
+      file_path = File.join(profile_dir, filename)
+      File.open(file_path, 'r') do |file|
+        profile_data = JSON.parse(File.read(file), symbolize_names: true)
+        profile_groups += profile_data[:profile][:groups]
+      end
+    end
+
+    # Initialize buckets
+    buckets = []
+    current_bucket = { id: 'bucket-1', total_time: 0, specs: [] }
+    bucket_index = 1
+
+    # Get a random distribution. Could iterate on this to pack bins
+    rng = Random.new(3)
+    profile_groups = profile_groups.shuffle(random: rng)
+
+    # Categorize specs into buckets
+    profile_groups.each do |group|
+      total_time = group[:total_time]
+
+      # Skip specs that run for less than 10 seconds
+      next if total_time < 10
+
+      if current_bucket[:total_time] + total_time > (60 * 40)
```

Comment:
is `60 * 40` 40 minutes?  Might be worth adding a comment or some variables to indicate what this is checking.